app/src/main/java/app/revanced/integrations/patches/LithoFilterPatch.java

```diff
@@ -119,73 +151,96 @@ public final class LithoFilterPatch {
 
     private final BlockRule[] rules;
 
     public ButtonsPatch() {
-        BlockRule like = new BlockRule(SettingsEnum.HIDE_LIKE_BUTTON, "|like_button");
-        dislikeRule = new BlockRule(SettingsEnum.HIDE_DISLIKE_BUTTON, "dislike_button");
-        BlockRule download = new BlockRule(SettingsEnum.HIDE_DOWNLOAD_BUTTON, "download_button");
-        actionButtonsRule = new BlockRule(SettingsEnum.HIDE_ACTION_BUTTON, "ContainerType|video_action_button");
-        BlockRule playlist = new BlockRule(SettingsEnum.HIDE_PLAYLIST_BUTTON, "save_to_playlist_button");
-        rules = new BlockRule[]{like, dislikeRule, download, actionButtonsRule, playlist};
-
-        actionBarRule = new BlockRule(null, "video_action_bar");
+        BlockRule like = new BlockRule(SettingsEnum.HIDE_LIKE_BUTTON, "CellType|ScrollableContainerType|ContainerType|ContainerType|like_button");
+        BlockRule dislike = new BlockRule(SettingsEnum.HIDE_DISLIKE_BUTTON, "CellType|ScrollableContainerType|ContainerType|ContainerType|dislike_button");
+        BlockRule share = new BlockRule(SettingsEnum.HIDE_SHARE_BUTTON, "yt_outline_share");
+        BlockRule live_chat = new BlockRule(SettingsEnum.HIDE_LIVE_CHAT_BUTTON, "yt_outline_message_bubble_overlap");
+        BlockRule report = new BlockRule(SettingsEnum.HIDE_REPORT_BUTTON, "yt_outline_flag");
+        BlockRule shorts = new BlockRule(SettingsEnum.HIDE_SHORTS_BUTTON, "yt_outline_youtube_shorts_plus");
+        BlockRule thanks = new BlockRule(SettingsEnum.HIDE_THANKS_BUTTON, "yt_outline_dollar_sign_heart");
+        BlockRule clip = new BlockRule(SettingsEnum.HIDE_CLIP_BUTTON, "yt_outline_scissors");
+        BlockRule download = new BlockRule(SettingsEnum.HIDE_DOWNLOAD_BUTTON, "CellType|ScrollableContainerType|ContainerType|ContainerType|download_button");
+        BlockRule playlist = new BlockRule(SettingsEnum.HIDE_PLAYLIST_BUTTON, "CellType|ScrollableContainerType|ContainerType|ContainerType|save_to_playlist_button");
+        rules = new BlockRule[]{
+            like,
+            dislike,
+            share,
+            download,
+            live_chat,
+            report,
+            shorts,
+            thanks,
+            clip,
+            playlist
+        };
 
         this.register.registerAll(
-                like,
-                dislikeRule,
-                download,
-                playlist
+            like,
+            dislike,
+            share,
+            download,
+            live_chat,
+            report,
+            shorts,
+            thanks,
+            clip,
+            playlist
         );
     }
 
-    private boolean hideActionBar() {
-        for (BlockRule rule : rules) if (!rule.isEnabled()) return false;
-        return true;
+    private boolean hideActionButtons(final String path, final ByteBuffer buffer) {
+        for (BlockRule rule : rules) {
+            if (rule.isEnabled()) {
+                if (Extensions.containsAny(path, actionBar.getBlocks()[0])) {
```

Comment:
Why not use `actionBar.check(path).isBlocked()` here?

Reply:
Maybe I did it out of distraction.

Reply:
EDIT: Fixed!